File: backend/src/main/scala/slaydemo/backend/battle/objects/BattleSurvivalOutcome.scala
package slaydemo.backend.battle.objects

enum BattleSurvivalOutcome {
  case Survived
  case Eliminated
}

object BattleSurvivalOutcome {
  def fromAliveAtEnd(aliveAtEnd: Boolean): BattleSurvivalOutcome =
    if aliveAtEnd then BattleSurvivalOutcome.Survived else BattleSurvivalOutcome.Eliminated

  def aliveAtEnd(value: BattleSurvivalOutcome): Boolean =
    value == BattleSurvivalOutcome.Survived
}

File: frontend/src/objects/battle/microservices/actors/objects/player/BattleSurvivalOutcome.ts
export type BattleSurvivalOutcome = "Survived" | "Eliminated";

export function battleSurvivalOutcomeFromAliveAtEnd(aliveAtEnd: boolean): BattleSurvivalOutcome {
  return aliveAtEnd ? "Survived" : "Eliminated";
}

export function battleSurvivalOutcomeAliveAtEnd(value: BattleSurvivalOutcome): boolean {
  return value === "Survived";
}

